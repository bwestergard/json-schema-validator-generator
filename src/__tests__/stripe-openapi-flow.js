// @flow
import path from 'path';
import generateFlow from 'jsvg/flow/generate.js';

const MODELS_TO_SNAPSHOT = [
  'account',
  'balance_transaction',
  'card',
  'charge',
  'coupon',
  'customer',
  'discount',
  'dispute',
];
describe('Stripe openapi', () => {
  // $FlowFixMe
  const spec = require(path.join(__dirname, '../../', 'openapi/openapi/spec2.json'));
  MODELS_TO_SNAPSHOT.forEach((model) => {
    it(`generates flow types for ${model}`, () => {
      expect(generateFlow(spec.definitions[model])).toMatchSnapshot();
    });
  });
});
