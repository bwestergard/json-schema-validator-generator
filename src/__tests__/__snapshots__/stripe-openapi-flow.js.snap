// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stripe openapi generates flow types for the entire spec 1`] = `
"type Account = {
  active_payment_methods: Array<mixed>,
  business_name?: string,
  business_url?: string,
  charges_enabled: boolean,
  country: string,
  debit_negative_balances: boolean,
  decline_charge_on: T0,
  default_currency: string,
  details_submitted: boolean,
  display_name?: string,
  email?: string,
  external_accounts: {
    data: Array<{[key: string]: mixed}>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  id: string,
  legal_entity: T1,
  managed: boolean,
  mcc?: string,
  metadata?: {[key: string]: mixed},
  object: string,
  orders?: T2,
  payout_schedule: T3,
  payout_statement_descriptor?: string,
  payouts_enabled: boolean,
  product_description?: string,
  statement_descriptor?: string,
  support_email?: string,
  support_phone?: string,
  timezone?: string,
  tos_acceptance: T4,
  verification: T5,
};
type BalanceTransaction = {
  amount: number,
  available_on: number,
  created: number,
  currency: string,
  description?: string,
  fee: number,
  fee_details: T6,
  id: string,
  net: number,
  object: string,
  source?: string,
  status: string,
  type: string,
};
type Card = {
  account?: string,
  address_city?: string,
  address_country?: string,
  address_line1?: string,
  address_line1_check?: string,
  address_line2?: string,
  address_state?: string,
  address_zip?: string,
  address_zip_check?: string,
  available_payout_methods?: Array<mixed>,
  brand: string,
  country?: string,
  currency?: string,
  customer?: string,
  cvc_check?: string,
  default_for_currency?: boolean,
  dynamic_last4?: string,
  estimated_availability?: string,
  exp_month: number,
  exp_year: number,
  fingerprint?: string,
  funding: string,
  google_reference?: string,
  id: string,
  last4: string,
  metadata: {[key: string]: mixed},
  name?: string,
  object: string,
  recipient?: string,
  three_d_secure?: {[key: string]: mixed},
  tokenization_method?: string,
};
type Charge = {
  amount: number,
  amount_authorized?: number,
  amount_captured: number,
  amount_refunded: number,
  application?: string,
  application_fee?: string,
  balance_transaction?: string,
  captured: boolean,
  card?: Card,
  created: number,
  currency: string,
  customer?: string,
  description?: string,
  destination?: string,
  dispute?: string,
  failure_code?: string,
  failure_message?: string,
  fraud_details?: {[key: string]: mixed},
  id: string,
  invoice?: string,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  on_behalf_of?: string,
  order?: string,
  outcome?: T7,
  paid: boolean,
  receipt_email?: string,
  receipt_number?: string,
  refunded: boolean,
  refunds: {
    data: Array<T8>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  review?: string,
  shipping?: T9,
  source_transfer?: string,
  statement_descriptor?: string,
  status: string,
  transfer?: string,
  transfer_group?: string,
};
type Coupon = {
  amount_off?: number,
  created: number,
  currency?: string,
  duration: string,
  duration_in_months?: number,
  id: string,
  livemode: boolean,
  max_redemptions?: number,
  metadata: {[key: string]: mixed},
  object: string,
  percent_off?: number,
  redeem_by?: number,
  times_redeemed: number,
  valid: boolean,
};
type Customer = {
  account_balance: number,
  alipay_accounts?: {
    data: Array<T10>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  bank_accounts?: {
    data: Array<T11>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  business_vat_id?: string,
  cards: {
    data: Array<Card>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  created: number,
  currency?: string,
  default_bank_account?: string,
  default_card?: string,
  default_source?: string,
  delinquent?: boolean,
  description?: string,
  discount?: Discount,
  email?: string,
  id: string,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  shipping?: T12,
  sources: {
    data: Array<{[key: string]: mixed}>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  subscriptions: {
    data: Array<T13>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
};
type Discount = {
  coupon: Coupon,
  customer?: string,
  end?: number,
  object: string,
  start: number,
  subscription?: string,
};
type Dispute = {
  amount: number,
  balance_transactions: BalanceTransaction,
  charge: string,
  created: number,
  currency: string,
  evidence: {[key: string]: mixed},
  evidence_details: {[key: string]: mixed},
  id: string,
  is_charge_refundable: boolean,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  reason: string,
  status: string,
};
type T0 = {
  avs_failure: boolean,
  cvc_failure: boolean,
};
type T1 = {
  additional_owners?: T14,
  address: T15,
  address_kana?: T16,
  address_kanji?: T16,
  business_name?: string,
  business_name_kana?: string,
  business_name_kanji?: string,
  business_tax_id_provided: boolean,
  business_vat_id_provided: boolean,
  dob: T17,
  first_name?: string,
  first_name_kana?: string,
  first_name_kanji?: string,
  gender?: string,
  last_name?: string,
  last_name_kana?: string,
  last_name_kanji?: string,
  maiden_name?: string,
  personal_address: T15,
  personal_address_kana?: T16,
  personal_address_kanji?: T16,
  personal_id_number_provided: boolean,
  phone_number?: string,
  ssn_last_4_provided: boolean,
  type?: string,
  verification: T18,
};
type T2 = {
  authorization: T19,
  channels: T20,
  create_order_hook_url?: string,
  order_management: T21,
  pay_order_hook_url?: string,
  shipping: T22,
  taxes: T23,
};
type T3 = {
  delay_days: number,
  interval: string,
  monthly_anchor?: number,
  weekly_anchor?: string,
};
type T4 = {
  date?: number,
  ip?: string,
  user_agent?: string,
};
type T5 = {
  disabled_reason?: string,
  due_by?: number,
  fields_needed: Array<mixed>,
};
type T6 = {
  amount: number,
  application?: string,
  currency: string,
  description?: string,
  type: string,
};
type T7 = {
  network_status?: string,
  reason?: string,
  risk_level?: string,
  rule?: string,
  seller_message?: string,
  type: string,
};
type T8 = {
  amount: number,
  balance_transaction?: string,
  charge?: string,
  created: number,
  currency: string,
  description?: string,
  id: string,
  metadata: {[key: string]: mixed},
  object: string,
  reason?: string,
  receipt_number?: string,
  status?: string,
};
type T9 = {
  address: T24,
  carrier?: string,
  name: string,
  phone?: string,
  tracking_number?: string,
};
type T10 = {
  created: number,
  customer?: string,
  fingerprint: string,
  id: string,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  object: string,
  payment_amount?: number,
  payment_currency?: string,
  reusable: boolean,
  used: boolean,
  username: string,
};
type T11 = {
  account?: string,
  account_holder_name?: string,
  account_holder_type?: string,
  address_city?: string,
  address_line1?: string,
  address_line2?: string,
  address_state?: string,
  address_zip?: string,
  allows_debits?: boolean,
  bank_name?: string,
  country: string,
  currency: string,
  customer?: string,
  customer_reference?: string,
  default_for_currency?: boolean,
  fingerprint?: string,
  id: string,
  last4: string,
  metadata?: {[key: string]: mixed},
  object: string,
  reusable?: boolean,
  routing_number?: string,
  status: string,
  used?: boolean,
};
type T12 = {
  address: T24,
  name: string,
  phone?: string,
};
type T13 = {
  account_balance?: number,
  application_fee_percent?: number,
  billing?: string,
  cancel_at_period_end: boolean,
  canceled_at?: number,
  created: number,
  current_period_end?: number,
  current_period_start?: number,
  customer: string,
  days_until_due?: number,
  discount?: Discount,
  ended_at?: number,
  id: string,
  items?: {
    data: Array<T25>,
    has_more: boolean,
    object: \\"list\\",
    total_count?: number,
    url: string,
  },
  livemode: boolean,
  max_occurrences?: number,
  metadata: {[key: string]: mixed},
  object: string,
  on_behalf_of?: string,
  plan?: T26,
  quantity?: number,
  retains_own_balance?: boolean,
  start: number,
  status: string,
  tax_percent?: number,
  trial_end?: number,
  trial_start?: number,
};
type T14 = {
  address: T15,
  dob: T17,
  first_name?: string,
  last_name?: string,
  verification: T18,
};
type T15 = {
  city?: string,
  country?: string,
  line1?: string,
  line2?: string,
  postal_code?: string,
  state?: string,
};
type T16 = {
  city?: string,
  country?: string,
  line1?: string,
  line2?: string,
  postal_code?: string,
  state?: string,
  town?: string,
};
type T17 = {
  day?: number,
  month?: number,
  year?: number,
};
type T18 = {
  details?: string,
  details_code?: string,
  document?: string,
  status: string,
};
type T19 = {
  issuer?: string,
  provider?: string,
  type: string,
};
type T20 = {
  twitter?: T27,
};
type T21 = {
  provider?: string,
  type: string,
};
type T22 = {
  amount?: number,
  currency?: string,
  description?: string,
  free_above?: number,
  from_address?: T24,
  from_name?: string,
  provider?: string,
  provider_url?: string,
  rates?: T28,
  type: string,
};
type T23 = {
  description?: string,
  provider?: string,
  provider_url?: string,
  rate?: number,
  type: string,
};
type T24 = {
  city?: string,
  country?: string,
  line1?: string,
  line2?: string,
  postal_code?: string,
  state?: string,
};
type T25 = {
  created: number,
  id: string,
  object: string,
  plan: T26,
  quantity: number,
};
type T26 = {
  amount: number,
  created: number,
  currency: string,
  id: string,
  interval: string,
  interval_count: number,
  livemode: boolean,
  metadata: {[key: string]: mixed},
  name: string,
  object: string,
  statement_descriptor?: string,
  trial_period_days?: number,
};
type T27 = {
  enabled: boolean,
  marketplace_id?: string,
  merchant_id?: string,
  oauth_token_provided: boolean,
  oauth_token_secret_provided: boolean,
  privacy_url?: string,
  sales_terms?: string,
  shipping_policy?: string,
  terms_url?: string,
  username: string,
};
type T28 = {
  amount: number,
  currency: string,
  description: string,
};"
`;
